application_manager/plugins/spark_generic/plugin.py

```diff
@@ -0,0 +1,359 @@
+            self._log("%s | Parameter missing in submission: %s, "
+                      "please check the config file" %
+                     (time.strftime("%H:%M:%S"), str(ke)))
+
+            self._log("%s | Finished application execution with error" %
+                (time.strftime("%H:%M:%S")))
+
+            self.update_application_state("Error")
+
+        except Exception as e:
+            self._log("%s | Unknown error, please report to administrators "
+                      "of WP3 infrastructure" % (time.strftime("%H:%M:%S")))
+
+            self._log("%s | Finished application execution with error" %
+                (time.strftime("%H:%M:%S")))
+
+            self.update_application_state("Error")
+
+
+    def get_application_time(self):
+        return self.application_time
+
+    def _hdfs_spark_execution(self, master, remote_hdfs, key_path, args,
+                              job_bin_url, main_class, dependencies, 
+                              spark_applications_ids, number_of_attempts):
+
+        job_exec_id = str(uuid.uuid4())[0:7]
+        self._log("%s | Job execution ID: %s" %
+            (time.strftime("%H:%M:%S"), job_exec_id))
+
+        # Defining params
+        local_path = '/tmp/spark-jobs/' + job_exec_id + '/'
+        remote_path = 'ubuntu@' + master + ':' + local_path
+
+        job_binary_path = hdfs.get_path(job_bin_url)
+        
+        # Create temporary job directories
+        self._log("%s | Create temporary job directories" %
+                      (time.strftime("%H:%M:%S")))
+        self._mkdir(local_path)
+
+        # Create cluster directories
+        self._log("%s | Creating cluster directories" %
+                (time.strftime("%H:%M:%S")))
+        remote.execute_command(master, key_path,
+                               'mkdir -p %s' % local_path)
+
+        # Get job binary from hdfs
+        self._log("%s | Get job binary from hdfs" %
+                (time.strftime("%H:%M:%S")))
+        remote.copy_from_hdfs(master, key_path, remote_hdfs,
+                              job_binary_path, local_path)
+
+        # Enabling event log on cluster
+        self._log("%s | Enabling event log on cluster" %
+               (time.strftime("%H:%M:%S")))
+        self._enable_event_log(master, key_path, local_path)
+
+        # Submit job
+        self._log("%s | Starting job" % 
+                      (time.strftime("%H:%M:%S")))
+
+        local_binary_file = (local_path + remote.list_directory(key_path, 
+                                                                master, 
+                                                                local_path))
+
+        spark_job = self._submit_job(master, key_path, main_class, 
+                                     dependencies, local_binary_file, args)
+
+        spark_app_id = spark.get_running_app(master, 
+                                             spark_applications_ids,
+                                             number_of_attempts)
+
```

Comment:
Does this plugin use controller or monitor services? Where are the starting calls?

Reply:
No. We decided this in our last meeting.